### PR TITLE
Sb 12991 . Added Logger in sunbird-user-service

### PR DESCRIPTION
**Pull request by anmol2302**
1: Reffered Link: https://howtodoinjava.com/log4j2/log4j2-properties-example/
2: change log4j.properties file properties to write logs in file.
3: ```[DEBUG] 2019-06-12 17:39:15.630 [application-akka.actor.default-dispatcher-2] defaultLogger - {"eid":"BELOG","ets":1560341355630,"mid":"Sunbird.1560341355630.60d2fb65-0554-4ee4-9ea9-bae8ac517aa5","ver":"1.0","context":{"pdata":{"ver":"1.0","id":"Sunbird"}},"edata":{"eks":{"level":"DEBUG","msgId":"1014eca1-8886-48e6-b056-ba4024549dfd","message":"OrgController:createOrg:method ended at 1560341355630"}}}```

### CODE REVIEW

File: user-org-service/app/controllers/BaseController.java [hunk 1]
```diff
@@ -23,13 +25,17 @@
   @Inject private HttpExecutionContext httpObject;
 
   public String getDummyResponse() {
-    String dummyResponse =
+      String dummyResponse =
         "{\"id\":\"api.user.200ok\",\"ver\":\"v1\",\"ts\":\"2019-01-17 16:53:26:286+0530\",\"params\":{\"resmsgid\":null,\"msgid\":\"8e27cbf5-e299-43b0-bca7-8347f7ejk5abcf\",\"err\":null,\"status\":\"success\",\"errmsg\":null},\"responseCode\":\"OK\",\"result\":{\"response\":{\"response\":\"SUCCESS\",\"errors\":[]}}}";
+    ProjectLogger.log(
+            "BaseController : getDummyResponse : "+ dummyResponse, LoggerEnum.INFO.name());
```
Comment (manzarul) [inline user-org-service/app/controllers/BaseController.java]: Use same pattern for all logger (className:methodName:message) don't put colon with different tab. 
Method entry and exist will be always in debug

File: user-org-service/app/controllers/BaseController.java [hunk 2]
```diff
@@ -23,13 +25,17 @@
...
     return dummyResponse;
   }
 
   public CompletionStage<Result> handelRequest() {
     Http.RequestBody requestBody = request().body();
+    ProjectLogger.log(
+            "BaseController :handelRequest : request reached to BaseController controller with request " + requestBody, LoggerEnum.INFO.name());
```
Comment (manzarul) [inline user-org-service/app/controllers/BaseController.java]: same as above , colon is applicable for all logger

File: user-org-service/app/controllers/healthmanager/HealthController.java [hunk 3]
```diff
@@ -24,6 +26,10 @@
      */
 
     public CompletionStage<Result> getHealth() {
+
+        ProjectLogger.log(
+                "HealthController :getHealth() : in HealthController ", LoggerEnum.INFO.name());
```
Comment (manzarul) [inline user-org-service/app/controllers/healthmanager/HealthController.java]: it should have method name only no brackets.
For method entry log should be :
("className:methodname:call started at timeinmilles",LoggerEnum.debug.name());
for method end:
("className:methodname:call ended at timeinmilles",LoggerEnum.debug.name());

File: user-org-service/conf/log4j2.properties [hunk 4]
```diff
@@ -1,16 +1,21 @@
-status = info
 name = PropertiesConfig
```
Comment (manzarul) [inline user-org-service/conf/log4j2.properties]: change PropertiesConfig to userLoggerConfig

File: user-org-service/conf/log4j2.properties [hunk 5]
```diff
@@ -1,16 +1,21 @@
...
-logger.default.level = info
-
-
+appender.console.layout.pattern = %d{dd.MM.yyyy HH:mm:ss,SSS} %-5p %c{1}:%L - %m%n
+#  --------------------------------------------------------------------------------------------
+# configuration to print logs to file
+appender.file.type = File
+appender.file.name = serviceLogs
+appender.file.fileName=${filename}
+appender.file.append="false"
```
Comment (manzarul) [inline user-org-service/conf/log4j2.properties]: make file appender as true.
appender.rolling.policies.size.size=10MB
appender.rolling.strategy.max = 10

File: user-org-util/src/main/java/org/sunbird/util/JsonKey.java [hunk 6]
```diff
@@ -0,0 +1,914 @@
+package org.sunbird.util;
```
Comment (manzarul) [inline user-org-util/src/main/java/org/sunbird/util/JsonKey.java]: change JsonKey to userOrgJsonKey
keep only those keys which are currently used.

File: user-org-util/src/main/java/org/sunbird/util/LoggerEnum.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package org.sunbird.util;
+
+/** @author Manzarul */
+public enum LoggerEnum {
+  INFO,
+  WARN,
+  DEBUG,
+  ERROR,
+  BE_LOG,
```
Comment (manzarul) [inline user-org-util/src/main/java/org/sunbird/util/LoggerEnum.java]: Change BELOG and PERFLOG

File: user-org-util/src/main/resources/log4j2.properties [hunk 8]
```diff
@@ -0,0 +1,21 @@
+name = PropertiesConfig
+property.filename = logs/sunbird-user-service.log
```
Comment (manzarul) [inline user-org-util/src/main/resources/log4j2.properties]: use only one place log4j2.properties file.

File: user-org-service/app/controllers/healthmanager/HealthController.java [hunk 9]
```diff
@@ -15,7 +17,9 @@
  */
 public class HealthController extends BaseController {
 
+
     private static final String userOrgServiceName = "user-org-service";
+    private static final String TAG="HealthController";
```
Comment (manzarul) [inline user-org-service/app/controllers/healthmanager/HealthController.java]: TAG to className

File: user-org-service/app/controllers/healthmanager/HealthController.java [hunk 10]
```diff
@@ -15,30 +13,32 @@
...
+    return response;
+  }
+
+  /**
+   * This action method is responsible to check user-service health
+   *
+   * @return a CompletableFuture of success response
+   */
+  public CompletionStage<Result> getUserOrgServiceHealth(String health) {
+    printProjectLogs(HealthController.class.getSimpleName(), "getHealth", true);
```
Comment (manzarul) [inline user-org-service/app/controllers/healthmanager/HealthController.java]: chnages getHealth to getUserOrgServiceHealth

File: user-org-service/app/controllers/healthmanager/HealthController.java [hunk 11]
```diff
@@ -15,30 +13,32 @@
...
+    CompletableFuture<String> cf = new CompletableFuture<>();
+    cf.complete(getDummyResponse());
+    printProjectLogs(HealthController.class.getSimpleName(), "getHealth", false);
```
Comment (manzarul) [inline user-org-service/app/controllers/healthmanager/HealthController.java]: change  getHealth to getUserOrgServiceHealth

**Review by manzarul — APPROVED** [review]
PR can be merged.

**Review by indrajra — CHANGES_REQUESTED** [review]
I dislike putting isBegin flags.

**Review by indrajra — APPROVED** [review]

**Pull request merged**